plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -134,13 +139,32 @@ private void updateModelGroup(
         if (Boolean.TRUE.equals(updateModelGroupInput.getIsAddAllBackendRoles())) {
             source.put(MLModelGroup.BACKEND_ROLES_FIELD, user.getBackendRoles());
         }
-        if (StringUtils.isNotBlank(updateModelGroupInput.getName())) {
-            source.put(MLModelGroup.MODEL_GROUP_NAME_FIELD, updateModelGroupInput.getName());
-        }
         if (StringUtils.isNotBlank(updateModelGroupInput.getDescription())) {
             source.put(MLModelGroup.DESCRIPTION_FIELD, updateModelGroupInput.getDescription());
         }
+        if (StringUtils.isNotBlank(updateModelGroupInput.getName()) && !updateModelGroupInput.getName().equals(modelGroupName)) {
+            mlModelGroupManager.validateUniqueModelGroupName(updateModelGroupInput.getName(), ActionListener.wrap(modelGroups -> {
+                if (modelGroups != null
+                    && modelGroups.getHits().getTotalHits() != null
+                    && modelGroups.getHits().getTotalHits().value != 0) {
+                    throw new IllegalArgumentException(
+                        "The name you provided is already being used by another model group. Please provide a different name"
```

Comment:
How about:

"The provided model group name: " + modelGroupName " + " already exists. Please provided a different model group name."